### ISSUE TITLE
IOS `pod install` is failing

### ISSUE THREAD

**Issue by sebqq** [OneSignal/react-native-onesignal#922]
Hello,

I have problem installing this library. After `pod install` command following error message is shown:

```
[!] CocoaPods could not find compatible versions for pod "OneSignal":
  In snapshot (Podfile.lock):
    OneSignal (< 3.0, = 2.12.2, >= 2.9.3)

  In Podfile:
    OneSignal (< 3.6.1, >= 2.9.3)

    react-native-onesignal (from `../node_modules/react-native-onesignal`) was resolved to 3.6.1, which depends on
      OneSignal (= 2.12.3)

Specs satisfying the `OneSignal (< 3.6.1, >= 2.9.3), OneSignal (< 3.0, = 2.12.2, >= 2.9.3), OneSignal (= 2.12.3)` dependency were found, but they required a higher minimum deployment target.
```

Don't you have any suggestions? Thank you so much!

react-native: 0.61.5
react-native-onesignal: 3.6.1

**Comment by sebqq** [comment]
Eveything is working fine with 3.6.0.

**Comment by rgomezp** [comment]
Can you try `pod update`?

**Comment by stefanoTron** [comment]
`pod update` worked for me. thanks

**Comment by sebqq** [comment]
Thank you so much! `pod update` is also working for me.

**Comment by GHO5TMEDO** [comment]
pod update worked for me too!

**Comment by smallfleet** [comment]
@GHO5TMEDO 

What is your React Native and OneSignal Version?

**Comment by rajeshsmallarc** [comment]
@GHO5TMEDO 
What is your React Native, Cocoa Pods and OneSignal Version?

**Comment by EsportudoAdmin** [comment]
`pod update` didn't work for me 🤨 @rgomezp I'm using the new Apple M1 chip..any ideas?

The message persists:

[!] CocoaPods could not find compatible versions for pod "OneSignal":
  In Podfile:
    OneSignal (< 3.0, >= 2.9.3)

    react-native-onesignal (from `../node_modules/react-native-onesignal`) was resolved to 4.0.3, which depends on
      OneSignal (= 3.1.0)

Specs satisfying the `OneSignal (< 3.0, >= 2.9.3), OneSignal (= 3.1.0)` dependency were found, but they required a higher minimum deployment target.

**Comment by djw27** [comment]
> `pod update` didn't work for me 🤨 @rgomezp I'm using the new Apple M1 chip..any ideas?
> 
> The message persists:
> 
> [!] CocoaPods could not find compatible versions for pod "OneSignal":
> In Podfile:
> OneSignal (< 3.0, >= 2.9.3)
> 
> ```
> react-native-onesignal (from `../node_modules/react-native-onesignal`) was resolved to 4.0.3, which depends on
>   OneSignal (= 3.1.0)
> ```
> 
> Specs satisfying the `OneSignal (< 3.0, >= 2.9.3), OneSignal (= 3.1.0)` dependency were found, but they required a higher minimum deployment target.

This looks as though you have

```
target 'OneSignalNotificationServiceExtension' do
  pod 'OneSignal', '>= 2.9.3', '< 3.0'
end
```

specified in your Podfile, which isn't valid for react-native-onesignal >= 4.0.0

**Comment by guiac** [comment]
did you fix it? I have same problem

**Comment by Apatoallo** [comment]
I have same problem, pod update didn't work for me too

**Comment by Git-Ashley** [comment]
also having this issue - has anyone found a fix for it yet?

**Comment by rgomezp** [comment]
try deintegrating the pods and reinstalling via

`pod deintegrate && pod install`

**Comment by sinclas** [comment]
Having the same issue: M1

CocoaPods could not find compatible versions for pod "OneSignal":
  In snapshot (Podfile.lock):
    OneSignal (= 2.12.6)

  In Podfile:
    react-native-onesignal (from `../node_modules/react-native-onesignal`) was resolved to 3.9.4, which depends on
      OneSignal (= 2.16.5)

Specs satisfying the `OneSignal (= 2.12.6), OneSignal (= 2.16.5)` dependency were found, but they required a higher minimum deployment target.  

Looks like I have to update a lot?

**Comment by mrinterestfull** [comment]
What is the final solution here?

yarn.lock
react-native-onesignal@^4.3.2:
  version "4.3.3"
  resolved "https://registry.yarnpkg.com/react-native-onesignal/-/react-native-onesignal-4.3.3.tgz#c2249b69787065cb8d361e273d4ff4fac3bbf5ae"
  integrity sha512-aJdgp32HsW1PEXgwMNXAaSbr037VdZmO70NPIQHWgCsIAGG4SD/MW9Kj197z1wdn/CRcfz/YNsZwhlmwUIgDMA==
  dependencies:
    invariant "^2.2.2"


Podfile.lock
 - react-native-onesignal (4.3.3):
    - OneSignalXCFramework (= 3.9.1)
    - React (< 1.0.0, >= 0.13.0)


or I also tried
target 'OneSignalNotificationServiceExtension' do
  pod 'OneSignal', '>= 2.9.3', '< 5.0'
  #pod 'OneSignalXCFramework', '>= 3.0.0', '< 5.0'
end


M1 processor. IOS target >11 react-native

**Comment by chaudev** [comment]
**On M1 Mac**

- Locate **Terminal.app** in Finder. _(Applications->Terminal.app)_
- Right-click and choose **Get Info**
- Check the **“Open using Rosetta”**
- Quit all instances of Terminal app and run it again
- Run **sudo gem install ffi**
- Run **pod update**
- Run **pod install**

**Comment by mrinterestfull** [comment]
Hello,
Just so we understand.
The above workaround and recommendation is to compile the app using emulator aka to emulate intel based processor (amd64); because I assume the library currently does not support arm64?

Are there plans to have arm64 support native without emulators?
Thank you
Lucas

**Comment by 0xtruly** [comment]
> Can you try `pod update`?

This worked for me too

**Comment by Ozaoujal** [comment]
running `pod update` from vscode didn't worked, while running it from terminal everything went right, m1 rn:v0.64.3

**Comment by VaibhavAcharya22** [comment]
I solved this error by changing the version of ios to `platform :ios, '12.0'` in `Podfile`.

**Comment by VaibhavAcharya22** [comment]
> Hello,
> 
> I have problem installing this library. After `pod install` command following error message is shown:
> 
> ```
> [!] CocoaPods could not find compatible versions for pod "OneSignal":
>   In snapshot (Podfile.lock):
>     OneSignal (< 3.0, = 2.12.2, >= 2.9.3)
> 
>   In Podfile:
>     OneSignal (< 3.6.1, >= 2.9.3)
> 
>     react-native-onesignal (from `../node_modules/react-native-onesignal`) was resolved to 3.6.1, which depends on
>       OneSignal (= 2.12.3)
> 
> Specs satisfying the `OneSignal (< 3.6.1, >= 2.9.3), OneSignal (< 3.0, = 2.12.2, >= 2.9.3), OneSignal (= 2.12.3)` dependency were found, but they required a higher minimum deployment target.
> ```
> 
> Don't you have any suggestions? Thank you so much!
> 
> react-native: 0.61.5 react-native-onesignal: 3.6.1

I solved this error by changing the version of ios to `platform :ios, '12.0'` in `Podfile`.

**Comment by SevdaCimen** [comment]
this solution works for me , 
https://github.com/OneSignal/react-native-onesignal/issues/922#issuecomment-1966085288
Thank you!

**Comment by vaibhav-simform** [comment]
> this solution works for me , 
> https://github.com/OneSignal/react-native-onesignal/issues/922#issuecomment-1966085288
> Thank you!

Your Welcome 👍🏻

**Comment by Med5KDB** [comment]
Same issue here. `pod update` not working for me. I got the same error when running both `pod install` and `pod update`. See attached screenshot.  I also change the `ios deployment target` on Xcode but not working.


Hey @VaibhavAcharya22! Thanks for your suggestion.
Would you mind to tell me from which version you changed to `12.0`? Following is what I have:
`platform :ios, podfile_properties['ios.deploymentTarget'] || '13.4'`. Should I downgrade to your given version i.e 12.0?

I've tried... it was working but got the similar error but w/ `React common/ Yoga` (not pretty sure for the name, could send a screenshot or entire error if needed). So I put back the old line...

<img width="1274" alt="Screenshot 2024-08-17 at 23 39 22" src="https://github.com/user-attachments/assets/3d852d56-9fe8-4797-ab86-0dd20edcc684">

Please let me know if you need more information or if there specific config I should check.

**Comment by nan-li** [comment]
Hi @Med5KDB you can also try to explicitly update the  OneSignalXCFramework
```
$ pod repo update
$ pod update OneSignalXCFramework
```

What react-native-onesignal version are you using?

**Comment by Med5KDB** [comment]
Hi @nan-li , thanks for your reply.
I'm using `5.2.1`.
I tried to update the OneSignalXCFramework as you suggested but still not working. Following is the kind of error (since it's in red) I'm getting:
`[!] No Podfile.lock' found in the project directory, run pod install'.` And as you can see the suggested solution is what lead to this issue😅

**Comment by nan-li** [comment]
`pod repo update` updates your local specs from cocoapods, did you then follow the suggestion to run `pod install` again?

**Comment by Med5KDB** [comment]
`pod repo update` works even if I got a notification telling that a newer version i.e `1.15.0`  of `CocoaPods` is available. The problem is w/ your second suggested command i.e `pod update OneSignalXCFramework` as I told you above.

**Comment by mujehoxe** [comment]
Any fixes yet, i am having this error, none of the above worked

```
[!] CocoaPods could not find compatible versions for pod "OneSignalXCFramework":
  In Podfile:
    OneSignalXCFramework (< 6.0, >= 5.0)

    react-native-onesignal (from `../node_modules/react-native-onesignal`) was resolved to 4.5.1, which depends on
      OneSignalXCFramework (= 3.12.4)

Specs satisfying the `OneSignalXCFramework (< 6.0, >= 5.0), OneSignalXCFramework (= 3.12.4)` dependency were found, but they required a higher minimum deployment target.
```
